src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,105 @@
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Entry<K, V>[] table;
+    private int size;
+    private int capacity = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+
+    private static class Entry<K, V> {
+        private K key;
+        private V value;
+        Entry<K, V> next;
+
+        private Entry(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+
+    public MyHashMap() {
+        this.table = new Entry[INITIAL_CAPACITY];
+    }
+
+    private int hashCode(K key) {
+        if (key == null) {
+            return 0;
+        } else {
+            return Math.abs(key.hashCode()) % table.length;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        int index = hashCode(key);
+        if (table[index] == null) {
+            table[index] = new Entry<>(key, value);
+            size++;
+            return;
+        }
+        if (index == 0) {
+            table[index].value = value;
+            return;
+        }
+
+        Entry<K, V> entry = table[index];
+        while (entry != null) {
+            if (entry.key.equals(key)) {
+                entry.value = value;
+                return;
+            } else {
+                entry = entry.next;
+            }
+        }
+        if (size++ > capacity) {
```

Comment:
```suggestion
        if (size++ > table.size * LOAD_FACTOR) {
```
and then we can remove lines: 13, 76

Reply:
метод rehash() краще за все викликати на початку або в кінці після/до того як додамо елемент

Reply:
fixed